src/objective-c/tests/UnitTests/GRPCCallOptionsTests.m

```diff
@@ -0,0 +1,161 @@
+static NSTimeInterval kGRPCCallOptionsTestKeepAliveTimeout = 120;
+static NSTimeInterval kGRPCCallOptionsTestKeepAliveInterval = 60;
+static NSTimeInterval kGRPCCallOptionsTestConnectMaxBackoff = 180;
+static NSTimeInterval kGRPCCallOptionsTestConnectMinTimeout = 210;
+static NSTimeInterval kGRPCCallOptionsTestConnectInitialBackoff = 12;
+static NSString *const kGRPCCallOptionsTestPEMPrivateKey = @"dummy_key";
+static NSString *const kGRPCCallOptionsTestPEMRootCertificates = @"dummy_cert";
+static NSString *const kGRPCCallOptionsTestPEMCertificateChain = @"dummy_chain";
+static GRPCTransportType kGRPCCallOptionsTestTransportType = GRPCTransportTypeChttp2BoringSSL;
+static GRPCTransportID kGRPCCallOptionsTestTransportID = "dummy_transport";
+static NSString *const kGRPCCallOptionsTestHostNameOverride = @"localhost";
+static NSString *const kGRPCCallOptionsTestChannelPoolDomain = @"dummy_domain";
+static NSUInteger kGRPCCallOptionsTestChannelID = 111;
+
+@interface GRPCCallOptionsTests : XCTestCase
+
+@end
+
+@implementation GRPCCallOptionsTests
+
+/** Verify a valid immutable copy can be created from GRPCCallOptions. */
+- (void)testCreateImmutableCopyFromImmutable {
+  GRPCCallOptions *opt = [[GRPCCallOptions alloc] init];
+  GRPCCallOptions *subject = [opt copy];
+  XCTAssertTrue([subject isKindOfClass:[GRPCCallOptions class]]);
+}
+
+/** Verify a valid mutable copy can be created from GRPCCallOptions. */
+- (void)testCreateMutableCopyFromImmutable {
+  GRPCCallOptions *opt = [[GRPCCallOptions alloc] init];
+  GRPCMutableCallOptions *subject = [opt mutableCopy];
+  XCTAssertTrue([subject isKindOfClass:[GRPCMutableCallOptions class]]);
+}
+
+/** Verify a valid mutable copy can be created from GRPCMutableCallOptions. */
+- (void)testCreateMutableCopyFromMutable {
+  GRPCMutableCallOptions *opt = [[GRPCMutableCallOptions alloc] init];
+  GRPCMutableCallOptions *subject = [opt mutableCopy];
+  XCTAssertTrue([subject isKindOfClass:[GRPCMutableCallOptions class]]);
+}
+
+/** Verify a valid immutable copy can be created from GRPCMutableCallOptions. */
+- (void)testCreateImmutableCopyFromMutable {
+  GRPCMutableCallOptions *mutableOpt = [[GRPCMutableCallOptions alloc] init];
+  GRPCCallOptions *subject = [mutableOpt copy];
+  XCTAssertTrue([subject isKindOfClass:[GRPCCallOptions class]]);
+}
+
+/** Verify property values are copied when copy from mutable options. */
+- (void)testCopyFromMutableCallOptions {
+  GRPCMutableCallOptions *mutableOpt = [self testMutableCallOptions];
+  XCTAssertTrue([self isEqualToTestCallOptions:[mutableOpt copy]]);
+  XCTAssertTrue([self isEqualToTestCallOptions:[mutableOpt mutableCopy]]);
+}
+
+/** Verify property values are copied when copy from immutable options */
+- (void)testCopyFromImmutableCallOptions {
+  GRPCCallOptions *opt = [[self testMutableCallOptions] copy];
+  XCTAssertTrue([self isEqualToTestCallOptions:[opt copy]]);
+  XCTAssertTrue([self isEqualToTestCallOptions:[opt mutableCopy]]);
+}
```

Comment:
suggestion:
In these assertions, it might be helpful to add assertions that output a representation of the object being tested in the assertion.

For example, if the assertion message printed `subject`, we might see that `subject` is nil or an instance of `UIView` and this might help developers identify the underlying cause of the failure more quickly without having to jump into the debugger.